Added light and dark mode. Added pricing cards

srushtisalke opened the pull request.

srushtisalke committed:
Added light and dark mode. Added pricing cards

srushtisalke commented:
@aina21  Could you please review this PR ?

shpomp commented on src/component/PriceCard.jsx:
```diff
@@ -0,0 +1,56 @@
+
+function PriceCard() {
+  return (
+    <div className="price-conteiner">
```
```suggestion
    <div className="price-container">
```
Typo in the "container" throughout some files.

srushtisalke replied:
Done.

shpomp commented on src/component/PriceCard.jsx:
```diff
@@ -0,0 +1,56 @@
+      <div className="price-row">
+        <div className="price-col">
+          <p>Basic</p>
+          <h3> 199dkk <span>/ month</span></h3>
```
Will this information come from the database or will always be set and hard-coded? 

Either way, such implementation is a bit too raw and difficult to maintain. 

Try to establish a data object and loop through it to render the list..

It would also be beneficial to have some different components (they can be here in the same file in this case) to render the different repetitive parts.
Refresh on React Thinking https://react.dev/learn/thinking-in-react and try to make this a bit more React-y 🙏🏽

srushtisalke replied:
Created new pricing card using data object and render.

shpomp commented on src/pages/Home.jsx:
```diff
@@ -4,10 +4,17 @@ import { apiPath } from "../api";
 const Home = () => {
   const [teamsDatabase, setTeamsDatabase] = useState([]);
+  const current_theme = localStorage.getItem('current_theme');  // This will store the default theme of web browser
+  const [theme, setTheme] = useState(current_theme ? current_theme : 'light'); // Created constant for dark mode
```
Comments are usually not needed for small and simple parts that are self-explanatory. 
(These days it can also signal that our friend robot was used, as it likes to comment explicitly 😅 )

srushtisalke replied:
Removed comments.

shpomp commented on src/styles/home.css:
```diff
@@ -56,4 +56,26 @@
+  background: #000000;
+}
+
+.dark .main-header {
```
These could be combine as you are applying just one and only style. 
You can just trail them one after another, separated by commas. 
```suggestion
.dark .main-header,  .dark .main-subheader, ... {
```

srushtisalke replied:
Done

shpomp commented on src/styles/pricecard.css:
```diff
@@ -0,0 +1,89 @@
+  cursor: pointer;
+  transition: background 0.5s;
+}
+
```
General note on CSS changes here:

As you are using some repetitive values when it concerns a theme, it would be good to establish some CSS variables https://www.w3schools.com/css/css3_variables.asp. 
That makes it easier to read and, most importantly, maintain or change.

shpomp reviewed (changes requested):
Good job! 💪🏽 ⭐ 

Left some notes and challenges. 

Additionally, for visual changes like this, screenshots or gifs in the PR should be included to help the potential reviewer get acquainted with the styling changes and variations. 
I did not run the code to see what it looks like, but from looking over the files it looks good! 

Ask if anything is unclear! 🙏🏽

srushtisalke committed:
Installed package for react icons.
Updated package.json and package-lock.json file.
Created Price-card
Added Poppins font inside index.html
Corrected "conteiner" spell to "container" in About, PriceCard, home, navbar and its styling components.
Added footer and styling
Updates to testimonials.

srushtisalke committed:
Updated PriceCard using data object and render.
Updated navbar links to respective components (about, pricing and conact).
Updated styling as well.

srushtisalke commented:
Hi @shpomp, 
I have made all changes as you suggested. They are in same pull request.
Please find video link here for demo: 
https://www.loom.com/share/5f1f0aa0cdc446c5ad6875a8d41fb5f4?sid=02882624-5819-4820-9a3b-8651c6811f5c

shpomp reviewed (approved):
👏⭐✨
